File: app/src/main/java/workshop/akbolatss/tagsnews/screen/board/BoardPresenter.java
package workshop.akbolatss.tagsnews.screen.board;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.observers.DisposableSingleObserver;
import io.reactivex.schedulers.Schedulers;
import workshop.akbolatss.tagsnews.base.BasePresenter;
import workshop.akbolatss.tagsnews.model.DBRssSourceRepository;
import workshop.akbolatss.tagsnews.model.dao.RssSource;

public class BoardPresenter extends BasePresenter<BoardView> {

    @Inject
    protected DBRssSourceRepository mRepository;

    @Inject
    public BoardPresenter() {
    }

    public void onLoadSources(final boolean isUpdating) {
        mRepository.getOnlyActive()
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe(new DisposableSingleObserver<List<RssSource>>() {
                    @Override
                    public void onSuccess(List<RssSource> rssSources) {
                        if (isUpdating) {
                            getView().onUpdateSources(rssSources);
                        } else {
                            getView().onInitSources(rssSources);
                        }
                    }

                    @Override
                    public void onError(Throwable e) {

                    }
                });
    }
}

File: app/src/main/java/workshop/akbolatss/tagsnews/screen/board/BoardPresenter.kt
package workshop.akbolatss.tagsnews.screen.board

import javax.inject.Inject

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.observers.DisposableSingleObserver
import io.reactivex.schedulers.Schedulers
import workshop.akbolatss.tagsnews.base.BasePresenter
import workshop.akbolatss.tagsnews.model.DBRssSourceRepository
import workshop.akbolatss.tagsnews.model.dao.RssSource

/**
 * MVP Presenter for #BoardActivity
 * @see BoardActivity
 */
class BoardPresenter @Inject constructor() : BasePresenter<BoardView>() {

    @Inject
    lateinit var mRepository: DBRssSourceRepository

    /**
     * Load list of active RSS sources from DB
     */
    fun onLoadSources(isUpdating: Boolean) {
        mRepository.onlyActive
                .observeOn(AndroidSchedulers.mainThread())
                .subscribeOn(Schedulers.io())
                .subscribe({ rssSources ->
                    if (isUpdating) {
                        view.onUpdateSources(rssSources)
                    } else {
                        view.onInitSources(rssSources)
                    }
                }, {
                })
    }
}
